src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,91 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationService registrationService = new RegistrationServiceImpl();
+
+    @AfterEach
+    void tearDown() {
+        registrationService = null;
+    }
+
+    @Test
+    void register_isLoginLongEnough_notOkay() {
+        User user = new User("cat", "1234567", 19);
+        assertThrows(FailedRegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_isCorrectAge_notOkay() {
+        User user1 = new User("cat1234", "1234569", 13);
+        User user2 = new User("cat4324", "2341233", 105);
+        assertThrows(FailedRegistrationException.class, () -> registrationService.register(user1));
+        assertThrows(FailedRegistrationException.class, () -> registrationService.register(user2));
+    }
+
+    @Test
+    void register_doesUserExist_notOkay() {
+        User user = new User("cat123456", "dodfmr12", 19);
+        Storage.PEOPLE.add(user);
+        assertThrows(FailedRegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_isLoginNull_notOkay() {
+        User user = new User(null, "kjniun123", 19);
+        assertThrows(FailedRegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_returnCorrectUser_okay() {
+        User expected = new User("kreep2004", "IloVeJava23", 19);
+        registrationService.register(expected);
+        User actual = new User(expected.getLogin(), expected.getPassword(), expected.getAge());
+        assertEquals(actual, expected);
+    }
+
+    @Test
+    void register_isUserNull_notOkay() {
+        User user = null;
+        assertThrows(FailedRegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void checkLogin_isLoginCorrect_notOkay() {

```

Comment:
this test verifies way too many scenarios - you should basically have separate tests for null, blank and incorrect length login